Makefile

```diff
@@ -1,39 +1,51 @@
+# Unless set otherwise, the container runtime is Docker
+DOCKER ?= docker
+
+prog ?= configurator
+sources := main.go $(wildcard cmd/*.go)
+plugin_source_prefix := pkg/generator/plugins
+plugin_sources := $(filter-out %_test.go,$(wildcard $(plugin_source_prefix)/*/*.go))
+plugin_binaries := $(addprefix lib/,$(patsubst %.go,%.so,$(notdir $(plugin_sources))))
 
 # build everything at once
+.PHONY: all
 all: plugins exe test
 
 # build the main executable to make configs
+.PHONY: main driver binaries exe
 main: exe
 driver: exe
 binaries: exe
-exe:
-	go build --tags=all -o configurator
+exe: $(prog)
 
+# build named executable from go sources
+$(prog): $(sources)
+    go build --tags=all -o $(prog)
 
-docker: binaries plugins
-	docker build . --build-arg REGISTRY_HOST=${REGISTRY_HOST} --no-cache --pull --tag '${NAME}:${VERSION}'
+.PHONY: container
+container: binaries plugins
+	$(DOCKER) build . --build-arg --no-cache --pull --tag 'configurator:testing'
```

Comment:
For this target and the `container-testing` target, we can replace `configurator` with `$(prog)`. Do we want to hard-code `testing` for the version? Or should we put do something like `<git_tag>-dirty`?

Reply:
I think we should do `<git_tag>-dirty` for `container` and then something like `$(prog):<git_tag>-dirty`. I think that would probably make more sense.